include/cantera/oneD/refine.h

```diff
@@ -69,51 +123,142 @@ class Domain1D;
+     *
+     * @param[in] n  Number of grid points
+     * @param[in] z  Point to array of grid points
+     * @param[in] x  Pointer to solution vector
+     *
+     * @returns The number of new grid points needed (size of #m_insertPts)
+     */
     int analyze(size_t n, const double* z, const double* x);
+
+    /**
+     * Constructs a new grid based on refinement locations determined by the analyze
+     * method.
+     *
+     * This function generates a new grid by inserting additional points into the
+     * current grid at locations where the `analyze` method has identified a need for
+     * refinement. The new grid points are placed midway between existing points deemed
+     * necessary for increased resolution. If no refinement is needed, the original
+     * grid is copied directly.
+     *
+     * @param[in] n The number of points in the original grid array `z`.
+     * @param[in] z Pointer to the array of original grid points.
+     * @param[in] nn The maximum number of points that the new grid array `znew` can hold.
+     * @param[out] znew Pointer to the array where the new grid points will be stored.
+     *
+     * @return The function returns 0 upon successful creation of the new grid. Throws
+     *         an exception if the provided output array size is insufficient to hold
+     *         the new grid.
+     *
+     * @deprecated Starting in %Cantera 3.1, this function is unused. To be removed
+     *             after %Cantera 3.1.
+     */
     int getNewGrid(int n, const double* z, int nn, double* znew);
+
+    /**
+     * Returns the number of new grid points that were needed.
+     */
     int nNewPoints() {
-        return static_cast<int>(m_loc.size());
+        return static_cast<int>(m_insertPts.size());
     }
+
+    /**
+     * Displays the results of the grid refinement analysis.
+     *
+     * This method logs information about where new grid points have been inserted
+     * and the components that required these insertions, if any. If no new points
+     * were needed, it logs that the current grid is sufficient.
+     *
+     * @note This method should be called after analyze() to report the outcomes of
+     * the refinement analysis.
+     */
     void show();
+
+    /**
+     * Returns true if a new grid point is needed to the right of grid index j.
+     *
+     * @param j  Grid point index
+     */
     bool newPointNeeded(size_t j) {
-        return m_loc.find(j) != m_loc.end();
+        return m_insertPts.find(j) != m_insertPts.end();
     }
+
+    /**
+     * Returns true if the grid point at index j should be kept.
+     *
+     * @param j Grid point index
+     */
     bool keepPoint(size_t j) {
-        return (m_keep[j] != -1);
+        return (m_keep[j] != REMOVE);
     }
-    double value(const double* x, size_t i, size_t j);
 
+    /**
+     * Returns the value of the solution component, n, at grid point j.
+     *
+     * @param x Solution vector
+     * @param n Solution Component index
```

Comment:
To avoid misleading links to the `Solution` class:
```suggestion
     * @param x %Solution vector
     * @param n %Solution component index
```